packages/flutter/lib/src/services/text_formatter.dart

```diff
@@ -172,29 +172,7 @@ class LengthLimitingTextInputFormatter extends TextInputFormatter {
     TextEditingValue newValue,
   ) {
     if (maxLength != null && maxLength > 0 && newValue.text.runes.length > maxLength) {
-      final TextSelection newSelection = newValue.selection.copyWith(
-          baseOffset: math.min(newValue.selection.start, maxLength),
-          extentOffset: math.min(newValue.selection.end, maxLength),
-      );
-      // This does not count grapheme clusters (i.e. characters visible to the user),
-      // it counts Unicode runes, which leaves out a number of useful possible
-      // characters (like many emoji), so this will be inaccurate in the
-      // presence of those characters. The Dart lang bug
-      // https://github.com/dart-lang/sdk/issues/28404 has been filed to
-      // address this in Dart.
-      // TODO(gspencer): convert this to count actual characters when Dart
-      // supports that.
-      final RuneIterator iterator = RuneIterator(newValue.text);
-      if (iterator.moveNext())
-        for (int count = 0; count < maxLength; ++count)
-          if (!iterator.moveNext())
-            break;
-      final String truncated = newValue.text.substring(0, iterator.rawIndex);
-      return TextEditingValue(
-        text: truncated,
-        selection: newSelection,
-        composing: TextRange.empty,
-      );
+      return oldValue;
```

Comment:
@gspencergoog @GaryQian Ok this is a dead simple way of doing this.  But why did we have all of this complexity in there before?  Is there any real behavior I'm breaking by making this change?  I'm interested in your thoughts.

Reply:
I can't remember why I implemented it that way, it may have had something to do with allowing the field to go over the maxLength when maxLength was not enforced, but if it's inconsistent with the way the native fields do it, then this is a good change to make.

Reply:
I think it's possible that I was trying to make it easier to convert later to counting grapheme clusters by implementing it the way I did.  I know that was a big consideration for that PR (#12648) at the time.

Reply:
I see, thanks for the explanation.

Proposal: I'll move forward with this PR, adding a some tests and a detailed comment similar to what was already here.  I'm likely going to be working on using Dart's new characters package to solve this grapheme cluster problem in the near future, so I'll also keep a reference to this old code for when I do.

I'll tag you for review when it's ready.